### ISSUE TITLE
Subscription Documentation

### ISSUE THREAD

**Issue by centrual** [j3k0/cordova-plugin-purchase#561]
Hello everybody,

I trying to figure out everything about subscription, but it's really really so hard to learn everything about subscription. Why we not write full subscription documentation?

### General Information
- [ ] What is the public key and where can i find?
- [ ] Is Apple or Android has query limit?
- [ ] What is the differences between ios and android?
- [ ] Trial subscriptions
### Database
- [ ] Database Structure and why? (Why always rdbms database is better for subscription jobs?)
### Validation & Security
- [ ] How we can make subscriptions secure?
- [ ] What is developer payload and why should i use it?
- [ ] How to validate subscription. ( Currently there is documentation for this )
- [ ] How we can handle original_transaction_id?
- [ ] How we can identify users only with transaction data? (For apps without login)
- [ ] How we can handle different devices using same google account?
### Advanced
- [ ] Why we need to check current subscriptions, every day? (for cancellation, renew etc.)
- [ ] How we can handle cancellations - refunds?
- [ ] How we can handle 100.000+ subscriptions? What is the server requirements for that?
- [ ] Howto make offline apps with subscriptions?

**Comment by achakra21** [comment]
Good Question Adding to this how to get subscription status if it active or expired?

**Comment by BlitZz96** [comment]
1+

**Comment by marcusgraf** [comment]
any update on this topic?

**Comment by saumya04** [comment]
Any Update??

**Comment by j3k0** [comment]
No update. Anyone has time to work on that?

**Comment by thesayyn** [comment]
actually i can work on this if it okay for you ?

**Comment by Dexus** [comment]
@thesayyn any contribution would be nice, so don't ask - just do it ;-)

**Comment by j3k0** [comment]
@thesayyn super ok for me ! :+1:

**Comment by znegva** [comment]
Hi.
Are there any news on this topic?

Where can I find the currently available info on this topic?
What I found was:
- the demo app [here](https://github.com/Fovea/cordova-plugin-purchase-demo/blob/master/www/js/index.js)
- a short example [here](https://github.com/j3k0/cordova-plugin-purchase/blob/master/doc/example.js) (not really much about subscriptions)

I think a main issue also is a missing working _example_ for a validation service (or a validation-function) where we can see what really should happen there.

**Comment by EvelinJK** [comment]
Bump. We have the same problem currently...there is no documentation about subscription validation with the plugin that is helpful by any means... it's really tiresome to figure the process out if you need to use your own server side validation..

**Comment by alexcroox** [comment]
General FAQ for me. iOS **requires** you to use their IAP system for recurring subscriptions. Reading the PlayStore terms, it seems this isn't a requirement if users can purchase/use their subscription on our normal website? 

I.e Android users can continue to use our stripe integration inside or Cordova, and we only need to use this plugin for iOS to avoid rejection.

**Comment by HarelM** [comment]
Thanks for this plugin! It work great!
I'm having trouble understanding the event when starting the app offline.
I have an auto-renewal subscription, so I'm using the `verify()` according the documentation with the fovea billing server.
When the phone is online and the app starts I get the owned event as expected and everything works fine.
But when the app start offline (I'm guessing) that the verification fails due to not internet connection.
What should I do in this case?
When starting the app, the `owned` status is `false` and it is turning to be `true` only when finish (I think) is completed, which happens after the verification, which won't happen if I'm offline.
My current solution is to store the "subscription is owned" in my state which is persistent if it ever changed to `true`.
But my concern is how to know that a subscription is now expired while allowing offline usage?
I don't mind that if the user is offline and the subscription just expired to allow him to continue the usage of subscription content, but I would like to know that one the user is online I know that the subscription has expired.
Code can be seen here if any one is interested:
https://github.com/IsraelHikingMap/Site/blob/master/IsraelHiking.Web/sources/application/services/purchase.service.ts

**Comment by j3k0** [comment]
Hi Harel.

You can implement your own offline strategy. 2 ideas.

1. Unlock when approved strategy:

- unlock features on "approved" and "owned" events.
- lock features back on "valid" and "expired" events.

2. Store the expiry date locally:

- localStorage or on a file to make it a bit harder to inject fake ones.
- in encrypted form (where only your server has the private key) to make it one step harder to fake.

As soon as you want to allow offline usage, there isn't any bullet proof strategy against fraud.

**Comment by HarelM** [comment]
@j3k0 Thanks for the super quick response!
Can you write the above suggestion using code examples? I have a hard time understanding the events registration strategy of this library...

I have no problem in validating the subscription when the user is online. I don't expect a full offline operational application, just to support the use-case of starting the app without internet connection, afterward I assume that at sometime the user will go online...
So basically, I need a code example of the `valid` and `expired` events I guess if it's not too much to ask...

P.S. I've sent you some e-mails but didn't got any response. I would greatly appreciate it if you could answer some of them, I really want to give back to this project.
